docs/pull_request.md

```diff
@@ -0,0 +1,158 @@
+# Pull requests
+
+## Introduction
+
+This document gives some clue about how to efficiently manage Pull Requests (PR). This document is a first draft and may be improved later.
+
+## Who should read this document?
+
+Every pull request reviewers, but also probably every ones who submit PRs.
+
+## Submitting PR
+
+### Who can submit pull requests?
+
+Basically every one who wants to contribute to the project! But there are some rules to follow.
+
+#### Humans
+
+Developers who are employed (or paid) by Element can directly clone the project and push their branches and create PR.
+
+External contributors must first fork the project and create PR to the mainline from there.
+
+##### Draft PR?
+
+Draft PR can be created when the submitter does not expect the PR to be reviewed and merged yet. It can be useful to publicly show the work, or to do a self-review first.
+
+Draft PR can also be created when it depends on other un-merged PR.
+
+In any case, it is better to explicitly declare in the description why the PR is a draft PR.
+
+##### Assigned to reviewer?
+
+For the moment we do not directly assign PR to ask for a review. In the future we could have some automation to do that. Anyway the core team should have a look to every PR, even when not digging into details. More reviewers, more eyes on the code change, more valuable feedback!
+
+##### When create split PR?
+
+To implement big new feature, it may be efficient to split the work into several smaller and scoped PRs. They will be easier to review, and they can be merged on develop faster.
+
+Big PR can take time, and there is a risk of future merge conflict.
+
+Feature flag can be used to avoid half implemented feature to be available in the application.
+
+That said, splitting into several PRs should not have the side effect to have more review to do, for instance if some code is added, then finally removed.
+
+##### Avoid unrelated fixing other issue in a big PR
+
+Each PR should focus on a single task. If other issues may be fixed when working in the area of it, it's preferable to open a dedicated PR.
```

Comment:
A good example I think is file formatting. If formatting more than a line or two the file formatting can really obscure the intended purpose of code changes in a PR. Opening a PR that just formats a group of related files can make reviewers lives much easier.

Reply:
I agree; I will add it

Reply:
formatting and noisy renames as separate PRs would be greatly appreciated :pray: